python/ray/data/_internal/logical/operators/read_operator.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Dict, Iterator
+
+import ray
+import ray.cloudpickle as cloudpickle
+from ray.data._internal.execution.interfaces import RefBundle, PhysicalOperator
+from ray.data._internal.execution.operators.map_operator import MapOperator
+from ray.data._internal.execution.operators.input_data_buffer import InputDataBuffer
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data.block import Block, BlockMetadata
+from ray.data.datasource.datasource import Reader
+
+
+class Read(LogicalOperator):
+    """Logical operator for read."""
+
+    def __init__(
+        self,
+        reader: Reader,
+        parallelism: int = -1,
+        ray_remote_args: Dict[str, Any] = None,
+        read_args: Dict[str, Any] = None,
+    ):
+        super().__init__("Read", [])
+        self._reader = reader
+        self._parallelism = parallelism
+        self._ray_remote_args = ray_remote_args
+        self._read_args = read_args
+
+    def get_physical_dag(self) -> PhysicalOperator:
+        """Get the corresponding DAG of physical operators for read."""
+        assert not self.input_dependencies, self.input_dependencies
+
+        read_tasks = self._reader.get_read_tasks(self._parallelism)
```

Comment:
Add a TODO to implement executing this in a remote task.

Btw, it's a bit weird we do task generation during planning. Is it possible to defer this to physical execution?

Reply:
Also +1 on this, read tasks are already the execution stuff.

Reply:
SG, moved the `Reader.get_read_tasks` into `InputDataBuffer`.